engine/src/main/java/org/hibernate/search/bridge/builtin/TikaBridge.java

```diff
@@ -40,28 +43,52 @@
  *
  * @author Hardy Ferentschik
  */
-public class TikaBridge implements MetadataProvidingFieldBridge {
+public class TikaBridge implements MetadataProvidingFieldBridge, AppliedOnTypeAwareBridge {
 	private static final Log log = LoggerFactory.make( MethodHandles.lookup() );
 
-	// Expensive, so only do it once. The Parser is threadsafe.
-	private final Parser parser = new AutoDetectParser();
+	private TikaParserProvider parserProvider;
+	private Parser parser;
 
 	private TikaMetadataProcessor metadataProcessor;
 	private TikaParseContextProvider parseContextProvider;
 
 	public TikaBridge() {
+		setParserProviderClass( null );
 		setMetadataProcessorClass( null );
 		setParseContextProviderClass( null );
 	}
 
+	@Override
+	public void setAppliedOnType(Class<?> returnType) {
```

Comment:
Don't know if it's already planned for 6 but having a proper hook would help :).

Reply:
It sure would. I didn't feel like adding such a think in 5... Having one interface per hook is already such a mess :)
In the Search 6 we don't really need a specific hook, because:

1. Binding aside, bridges are immutable. Parameters, if any, are processed by the bridge builder, and the builder can execute such initialization code in the build method (or you can execute it in the bridge constructor).
2. If push comes to shove, you already have a bind() method where you can execute such initialization code.

Reply:
Ok, cool, just wanted to check we were covered for 6!